core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -1312,6 +1315,23 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
           InferTypes.FIRST_KNOWN,
           OperandTypes.INTEGER_INTEGER.or(OperandTypes.BINARY_BINARY)
               .or(OperandTypes.UNSIGNED_NUMERIC_UNSIGNED_NUMERIC));
+  // Both operands should support bitwise operations
+
+  /**
+   * <code>{@code &}</code> operator.
+   */
+  public static final SqlBinaryOperator BITAND_OPERATOR =
+      new SqlBinaryOperator(
+          "&",
+          SqlKind.BITAND,
+          50,        // Higher precedence than XOR but lower than multiplication
+          true,
+          ReturnTypes.LARGEST_INT_BINARY_OR_LEAST_RESTRICTIVE,
+          InferTypes.FIRST_KNOWN,
+          OperandTypes.INTEGER_INTEGER.or(OperandTypes.BINARY_BINARY)
+              .or(OperandTypes.UNSIGNED_NUMERIC_UNSIGNED_NUMERIC)
+              .or(OperandTypes.family(SqlTypeFamily.UNSIGNED_NUMERIC, SqlTypeFamily.INTEGER)
+                  .or(OperandTypes.family(SqlTypeFamily.INTEGER, SqlTypeFamily.UNSIGNED_NUMERIC))));
```

Comment:
Clearly, BITAND and BITAND_OPERATOR should have the same behavior.
But I was asking whether the type checking rules match the other bitwise operators, such as BITXOR and BITOR.

Reply:
Can you also please answer this question?
What do you think we should do about the fact that BITXOR_OPERATOR has a different return type inference and a different operandTypeChecker?
One of the two must be wrong.
(I am not asking necessarily for a fix in this PR, I just want to understand what you think should be the strategy to align them).